File: multi_lang/node_templates/api_template.js
// Template: Express API skeleton (raw). Placeholders substituted by NodeApiGenerator.
const express = require('express');
const app = express();

app.use(express.json());

app.get(`/api/v1/__ROUTE__`, (req, res) => {
  res.json({ service: '__MODULE_NAME__', items: [] });
});

app.get('/healthz', (req, res) => {
  res.json({ status: 'ok' });
});

module.exports = app;

File: multi_lang/rust_templates/api_template.rs
// Template: Actix API skeleton (raw). Placeholders substituted by RustApiGenerator.
use actix_web::{get, web, App, HttpResponse, HttpServer, Responder};

async fn healthz() -> impl Responder {
    HttpResponse::Ok().json(serde_json::json!({ "status": "ok" }))
}

#[get("/api/v1/__ROUTE__")]
async fn list__MODEL_NAME__() -> impl Responder {
    HttpResponse::Ok().json(serde_json::json!({ "service": "__MODULE_NAME__", "items": [] }))
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| App::new().route("/healthz", web::get().to(healthz)).service(list__MODEL_NAME__))
        .bind(("127.0.0.1", 8080))?
        .run()
        .await
}